src/lib/y2storage/proposal_settings.rb

```diff
@@ -441,20 +442,39 @@ def load_features
       load_encryption
     end
 
+    def load_encryption_hash(enc)
+      passwd = enc["password"]
+      self.encryption_password = passwd if !passwd.nil? && !passwd.empty?
+
+      type = enc["type"]
+      # default luks1 is already set
+      self.encryption_method = EncryptionMethod::LUKS2 if !type.nil? && type == "luks2"
+
+      pbkdf = enc["pbkdf"]
+      self.encryption_pbkdf = Y2Storage::PbkdFunction.find(pbkdf) if !pbkdf.nil? && !pbkdf.empty?
+    end
+
     # Loads the default encryption settings
     #
-    # The encryption settings are not part of control.xml, but can be injected by a previous step of
-    # the installation, eg. the dialog of the Common Criteria system role
+    # The encryption settings are not part of the offical control.xml, but can be injected by a
+    # previous step of the installation, eg. the dialog of the Common Criteria system role or
+    # expert dialog
     def load_encryption
       enc = feature(:proposal, :encryption)
-
       return unless enc
-      return unless enc.respond_to?(:password)
 
-      passwd = enc.password.to_s
-      return if passwd.nil? || passwd.empty?
+      if enc.is_a?(Hash)
```

Comment:
I have just started the review, but this smells bad. The fact that `feature(:proposal, :encryption)` can return different data types depending on who decided the settings looks quite hacky.

Reply:
After a closer inspection, now I see why you did it. The way in which the Common Criteria role injects the password into the settings is certainly hacky.

That being said, I fear injecting the password as a plain value is also quite a bad idea.